WebAppForSolocoProject/Controllers/HomeController.cs

```diff
@@ -53,16 +58,109 @@ public IActionResult Create(HomeCreateVM model)
                             Directory.CreateDirectory(pathToCreate);
                             pathToCreate += " - directory succesfully created.";
                         }
-                        model.CreatedPaths.Add(pathToCreate);
+                        model.Logs.Add(pathToCreate);
                     }
                     return View("Create", model);
                 }
                 catch (UnauthorizedAccessException)
                 {
-                    return BadRequest("You don't have access in selected directory. Please change your base path.");
+                    model.Logs.Add("You don't have access in selected directory. Please change your base path.");
                 }
-            }    
+            }
             return View("Create",model);
         }
+
+        public IActionResult ChangeBasePath(HomeCreateVM model)
+        {
+            UpdateModel(model);
+
+            Thread thread = new Thread(new ThreadStart(ThreadMethod));
```

Comment:
Instead of create thread  which is join make this methode async and use await for Task which will run your ThreadMethode

Reply:
I need use thread and set the ApartmentState to STA, otherwise i got exception when invoke ShowDialog method on FolderBrowerDialog, but i changed action to be async